src/main/java/org/mitre/synthea/world/concepts/healthinsurance/CoverageRecord.java

```diff
@@ -19,111 +18,25 @@
-    public String ownerName;
-    private BigDecimal healthcareExpenses = Claim.ZERO_CENTS;
-    private BigDecimal coveredExpenses = Claim.ZERO_CENTS;
-    private BigDecimal insuranceCosts = Claim.ZERO_CENTS;
-    public BigDecimal remainingDeductible = Claim.ZERO_CENTS;
-
-    /**
-     * Create a new Plan with the given Payer.
-     * @param time The time the plan starts.
-     * @param plan The plan associated with the PlanRecord.
-     */
-    public PlanRecord(long time, InsurancePlan plan) {
-      this.start = time;
-      Calendar c = Calendar.getInstance();
-      c.setTimeInMillis(time);
-      c.add(Calendar.YEAR, 1);
-      this.stop = c.getTimeInMillis();
-      this.plan = plan;
-      this.remainingDeductible = plan.getDeductible();
-    }
-
-    /**
-     * Pay monthly premiums associated with this plan.
-     * @return  Cost of the premiums.
-     */
-    public BigDecimal payMonthlyPremiums() {
-      BigDecimal premiumPrice = (this.plan.payMonthlyPremium())
-          .add(this.secondaryPlan.payMonthlyPremium());
-      this.insuranceCosts = this.insuranceCosts.add(premiumPrice);
-      return premiumPrice;
-    }
-
-    public void updateStopTime(long updatedStopTime) {
-      this.stop = updatedStopTime;
-    }
-
-    @Override
-    public String toString() {
-      StringBuilder sb = new StringBuilder();
-      sb.append("[PlanRecord:");
-      sb.append(" Start: " + start);
-      sb.append(" Stop: " + stop);
-      sb.append(" Payer: " + this.plan.getPayer().getName() + "]");
-      return sb.toString();
-    }
-
-    public void incrementPatientExpenses(BigDecimal expenses) {
-      this.healthcareExpenses = this.healthcareExpenses.add(expenses);
-      this.plan.addUncoveredCost(expenses);
-    }
-
-    public void incrementPrimaryCoverage(BigDecimal coverage) {
-      this.coveredExpenses = this.coveredExpenses.add(coverage);
-      this.plan.addCoveredCost(coverage);
-    }
-
-    public void incrementSecondaryCoverage(BigDecimal coverage) {
-      this.coveredExpenses = this.coveredExpenses.add(coverage);
-      this.secondaryPlan.addCoveredCost(coverage);
-    }
-
-    public BigDecimal getHealthcareExpenses() {
-      return this.healthcareExpenses;
-    }
-
-    public BigDecimal getCoveredExpenses() {
-      return this.coveredExpenses;
-    }
-
-    public BigDecimal getInsuranceCosts() {
-      return this.insuranceCosts;
-    }
-
-    public boolean isDedctiblePlan() {
-      return this.plan.getDeductible().compareTo(BigDecimal.ZERO) == 1;
-    }
-  }
-
   @JSONSkip
   private Person person;
-  private List<PlanRecord> planHistory;
+  private Set<PlanRecord> planHistory;
+  private Long nextEnrollmentPeriod = null;
 
   /**
    * Create a new CoverageRecord for the given Person.
    * @param person The person.
    */
   public CoverageRecord(Person person) {
     this.person = person;
-    this.planHistory = new ArrayList<PlanRecord>();
+    this.planHistory = new HashSet<PlanRecord>();
```

Comment:
I'm not sure if changing from `List` to `Set` is the cause, but this PR breaks reproducibility.

I ran the following command twice, and found 3 patients (out of 100) that did not match. 

```
./run_synthea -s 11 -cs 22 -r 20230101 -e 20230101 -p 100
```

All of the differences were related to coverage. For instance,

```diff
      "insurance": [ {
        "sequence": 1,
        "focal": true,
        "coverage": {
+          "display": "Aetna"
-          "display": "Anthem"
        }
      } ],
```